Data race в NActors::ChangeActivity

Выявляется на нескольких тестах под tsan. Например тут: https://github.com/ydb-platform/ydb/actions/runs/13252177187
```
E   WARNING: ThreadSanitizer: data race (pid=158435)
E   Read of size 4 at 0x000034ce3cc8 by thread T98:
E   #0 NActors::ChangeActivity(long, unsigned int&, unsigned int&) /-S/ydb/library/actors/core/activity_guard.cpp:11:90 (ydbd+0x17d1a539) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #1 TInternalActorTypeGuard /-S/ydb/library/actors/core/activity_guard.h:33:13 (ydbd+0x17d42250) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #2 void NActors::TGenericExecutorThreadCtx::Spin<NActors::TExecutorThreadCtx, NActors::EThreadState>(unsigned long, std::__y1::atomic<bool>*) /-S/ydb/library/actors/core/executor_thread_ctx.cpp:24:82 (ydbd+0x17d42250)
E   #3 Spin /-S/ydb/library/actors/core/executor_thread_ctx.h:80:26 (ydbd+0x17d41759) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #4 NActors::TExecutorThreadCtx::Wait(unsigned long, std::__y1::atomic<bool>*) /-S/ydb/library/actors/core/executor_thread_ctx.cpp:85:13 (ydbd+0x17d41759)
E   #5 NActors::TBasicExecutorPool::GetReadyActivationCommon(unsigned long) /-S/ydb/library/actors/core/executor_pool_basic.cpp:246:43 (ydbd+0x17d276a1) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #6 NActors::TBasicExecutorPool::GetReadyActivation(unsigned long) /-S/ydb/library/actors/core/executor_pool_basic.cpp:292:20 (ydbd+0x17d27e10) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #7 NActors::TExecutorThread::ProcessExecutorPool() /-S/ydb/library/actors/core/executor_thread.cpp:502:43 (ydbd+0x17d49ff8) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #8 NActors::TExecutorThread::ThreadProc() /-S/ydb/library/actors/core/executor_thread.cpp:538:9 (ydbd+0x17d4b292) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #9 void* (anonymous namespace)::ThreadProcWrapper<ISimpleThread>(void*) /-S/util/system/thread.cpp:383:45 (ydbd+0x16953172) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #10 (anonymous namespace)::TPosixThread::ThreadProxy(void*) /-S/util/system/thread.cpp:244:20 (ydbd+0x1695364c) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   
E   Previous write of size 4 at 0x000034ce3cc8 by thread T52:
E   #0 NActors::ChangeActivity(long, unsigned int&, unsigned int&) /-S/ydb/library/actors/core/activity_guard.cpp:11:23 (ydbd+0x17d1a554) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #1 ~TInternalActorTypeGuard /-S/ydb/library/actors/core/activity_guard.h:48:13 (ydbd+0x17d423f2) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #2 void NActors::TGenericExecutorThreadCtx::Spin<NActors::TExecutorThreadCtx, NActors::EThreadState>(unsigned long, std::__y1::atomic<bool>*) /-S/ydb/library/actors/core/executor_thread_ctx.cpp:50:5 (ydbd+0x17d423f2)
E   #3 Spin /-S/ydb/library/actors/core/executor_thread_ctx.h:80:26 (ydbd+0x17d41759) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #4 NActors::TExecutorThreadCtx::Wait(unsigned long, std::__y1::atomic<bool>*) /-S/ydb/library/actors/core/executor_thread_ctx.cpp:85:13 (ydbd+0x17d41759)
E   #5 NActors::TBasicExecutorPool::GetReadyActivationCommon(unsigned long) /-S/ydb/library/actors/core/executor_pool_basic.cpp:246:43 (ydbd+0x17d276a1) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #6 NActors::TBasicExecutorPool::GetReadyActivation(unsigned long) /-S/ydb/library/actors/core/executor_pool_basic.cpp:292:20 (ydbd+0x17d27e10) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #7 NActors::TExecutorThread::ProcessExecutorPool() /-S/ydb/library/actors/core/executor_thread.cpp:502:43 (ydbd+0x17d49ff8) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #8 NActors::TExecutorThread::ThreadProc() /-S/ydb/library/actors/core/executor_thread.cpp:538:9 (ydbd+0x17d4b292) (BuildId: 444d3fb6c21b107aa3e72c9806b54953e2b4672b)
E   #9 void* (anonymous namespace)::ThreadProcWrapper<ISimpleThread>(void*) /-S/util/system/thread.cpp...
```

--- a/ydb/library/actors/core/activity_guard.h
+++ b/ydb/library/actors/core/activity_guard.h
@@ -45,7 +45,8 @@ public:
 
     ~TInternalActorTypeGuard() {
         if (Allowed) {
-            ChangeActivity(GetCycleCountFast(), Index, NextIndex);
+            ui32 prevIndex = Index;
+            ChangeActivity(GetCycleCountFast(), prevIndex, NextIndex);
         }
     }
 

